### ISSUE TITLE
skip_serializing_if internal tag

### ISSUE THREAD

**Issue by antoyo** [serde-rs/serde#1985]
Hi.
I'm not sure if it is already supported, but it would be great if there was a way to conditionally skip the serialization of the tag of an `enum`.
Is there already a way to do that?
Thanks.

**Comment by ColinTimBarndt** [comment]
You could use a wrapper struct that has a custom serializer implemented (see #2009)

**Comment by dtolnay** [comment]
Depending on what granularity you want to control whether to skip or not skip the tags, you could either wrap individual serializable enums as suggested in Colin's link, or wrap the whole `Serializer` to make it serialize *all* tagged enums in the whole input in untagged form instead. Something like:

```rust
let value = /*thing to serialize*/;
let mut ser = serde_json::Serializer::new(...);
value.serialize(AllEnumsUntagged(&mut ser))?;
```
